pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -189,10 +179,8 @@ private void autoComplete(int slashPosition, String input) {
             }
         }
         autoCompletePopup.getItems().setAll(resultToDisplay);
-        autoCompletePopup.show(xpathExpressionArea, xpathExpressionArea.getCharacterBoundsOnScreen(positionFrom, positionTo)
-                                                                       .get()
-                                                                       .getMaxX(), xpathExpressionArea
-                                   .getCharacterBoundsOnScreen(positionFrom, positionTo).get().getMaxY());
+        autoCompletePopup.show(xpathExpressionArea, xpathExpressionArea.getCharacterBoundsOnScreen(slashPosition, slashPosition + input.length()).get().getMaxX(),
```

Comment:
Again, you're calling getCharacterBoundsOnScreen twice. Call it once and use Optional::ifPresent because the `get()` call is unsage